pkg/sql/logictest/testdata/logic_test/mixed_version_can_login

```diff
@@ -21,7 +21,7 @@ SELECT crdb_internal.create_tenant(1001)
 upgrade 1
 
 query B
-SELECT crdb_internal.node_executable_version() SIMILAR TO '22.2-%'
+SELECT crdb_internal.node_executable_version() SIMILAR TO '23.1-%'
```

Comment:
```suggestion
SELECT crdb_internal.node_executable_version() SIMILAR TO '23.1'
```

Reply:
These suggestion should fix the tests. I have been debating with myself whether to use `SIMILAR TO '23.1'` vs ` = '23.1'`.
The latter is stricter, but the former generates a smaller diff. 🤷